cnf-tests/testsuites/pkg/features/features.go

```diff
@@ -0,0 +1,266 @@
+package features
+
+import (
+	"fmt"
+
+	sriovClean "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/clean"
+	sriovNamespaces "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/e2esuite/fec"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/e2esuite/ptp"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/e2esuite/sctp"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/e2esuite/security"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/e2esuite/sro"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/e2esuite/vrf"
+	testclient "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	testutils "github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/utils"
+	numaserialconf "github.com/openshift-kni/numaresources-operator/test/e2e/serial/config"
+	perfUtils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	perfClean "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/clean"
+	ptpClean "github.com/openshift/ptp-operator/test/utils/clean"
+)
+
+var (
+	performanceNS    = perfUtils.NamespaceTesting
+	dpdkNS           = namespaces.DpdkTest
+	sroNS            = namespaces.SroTestNamespace
+	gatekeeperNS     = testutils.GatekeeperTestingNamespace
+	sctpNS           = sctp.TestNamespace
+	vrfNS            = vrf.TestNamespace
+	sriovNS          = sriovNamespaces.Test
+	xtu32NS          = namespaces.XTU32Test
+	ovsqosNS         = namespaces.OVSQOSTest
+	tuningcniNS      = security.TestNamespace
+	tuningcniSriovNS = security.SriovTestNamespace
+	bondcniNS        = namespaces.BondTestNamespace
+)
+
+type SuiteManager interface {
+	BeforeSuite() error
+	AfterSuite() error
+}
+
+type PerformanceManager struct {
+}
+
+func (p *PerformanceManager) BeforeSuite() error {
+	return namespaces.Create(p.ns(), testclient.Client)
+}
+
+func (p *PerformanceManager) AfterSuite() error {
+	if !discovery.Enabled() {
+		perfClean.All()
+	}
+
+	return namespaces.Delete(p.ns(), testclient.Client)
+}
+
+func (p *PerformanceManager) ns() string {
+	return performanceNS
+}
+
+type DpdkManager struct {
+}
+
+func (p *DpdkManager) BeforeSuite() error {
+	return namespaces.Create(p.ns(), testclient.Client)
+}
+
+func (p *DpdkManager) AfterSuite() error {
+	return namespaces.Delete(p.ns(), testclient.Client)
+}
+
+func (p *DpdkManager) ns() string {
+	return dpdkNS
```

Comment:
Why can't you just hardcode the ns above? Same for the others.

Reply:
you mean in the ns function? i can, but i thought it will be easier to change when needed if all the namespaces are above in a list.

Reply:
not sure I understood what it'd be simpler

Reply:
instead old looking at this huge file. i can change that. not critical.

Reply:
The thing I was talking about is the ns() function. Why would we need it?

Reply:
because we use the ns for some of the features twice (in before and after), so when changing the ns we will have to change it in two places instead of just changing it once in the var.

Reply:
how comes, if we have the list of constants above? And even if it was simpler, it can be a field of the manager instead of a method.

Reply:
sorry, was talking about the constants